Makefile

```diff
@@ -242,31 +238,32 @@ $(imagec): $(call godeps,cmd/imagec/*.go) $(portlayerapi-client)
 	@$(GO) build $(RACE) -o ./$@ ./$(dir $<)
 
 
-$(docker-engine-api): $(call godeps,apiservers/engine/server/*.go) $(portlayerapi-client)
+$(docker-engine-api): $(call godeps,cmd/docker/*.go) $(portlayerapi-client)
 ifeq ($(OS),linux)
 	@echo Building docker-engine-api server...
-	@$(GO) build $(RACE) -o $@ ./apiservers/engine/server
+	@$(GO) build $(RACE) -o $@ ./cmd/docker
 else
 	@echo skipping docker-engine-api server, cannot build on non-linux
 endif
 
 # Common portlayer dependencies between client and server
-PORTLAYER_DEPS ?= apiservers/portlayer/swagger.yml \
-				  apiservers/portlayer/restapi/configure_port_layer.go \
-				  apiservers/portlayer/restapi/options/*.go apiservers/portlayer/restapi/handlers/*.go
+PORTLAYER_DEPS ?= lib/apiservers/portlayer/swagger.yml \
+				  lib/apiservers/portlayer/restapi/configure_port_layer.go \
+				  lib/apiservers/portlayer/restapi/options/*.go \
+				  lib/apiservers/portlayer/restapi/handlers/*.go
 
 $(portlayerapi-client): $(PORTLAYER_DEPS)  $(SWAGGER)
 	@echo regenerating swagger models and operations for Portlayer API client...
-	@$(SWAGGER) generate client -A PortLayer --template-dir templates  -t $(realpath $(dir $<)) -f $<
+	@$(SWAGGER) generate client -A PortLayer --template-dir lib/apiservers/templates  -t $(realpath $(dir $<)) -f $<
 
 
 $(portlayerapi-server): $(PORTLAYER_DEPS) $(SWAGGER)
 	@echo regenerating swagger models and operations for Portlayer API server...
-	@$(SWAGGER) generate server -A PortLayer -t $(realpath $(dir $<)) -f $<
+	@$(SWAGGER) generate server -A PortLayer --template-dir lib/apiservers/templates -t $(realpath $(dir $<)) -f $<
 
-$(portlayerapi): $(call godeps,apiservers/portlayer/cmd/port-layer-server/*.go) $(portlayerapi-server) $(portlayerapi-client)
+$(portlayerapi): $(call godeps,lib/apiservers/portlayer/cmd/port-layer-server/*.go) $(portlayerapi-server) $(portlayerapi-client)
```

Comment:
I would have moved the whole apiservers/ folder under cmd/ . None of the code under apiservers is being used by any other component, so it doesn't make sense to me to put it in lib.